crates/sui-core/src/execution_engine.rs

```diff
@@ -459,7 +459,7 @@ fn pay<S>(
 ) -> Result<(), ExecutionError> {
     check_recipients(&recipients, &amounts)?;
     let (mut coins, coin_type) = check_coins(&coin_objects, None)?;
-    let (total_coins, total_amount) = check_total_coins(&coins, &amounts)?;
+    let (_total_coins, _total_amount) = check_total_coins(&coins, &amounts)?;
```

Comment:
I remember the warnings were gone already with `cargo build -r`, is it warning from some other cmds?

Reply:
The related PR is https://github.com/MystenLabs/sui/pull/5555

Reply:
debug_assert_eq causes the warning, if the build is not in test/debug mode the variable used by `debug_assert_eq` will appear to be a unused variable

Reply:
if the warning is gone already we can close this PR, I will double check later today

Reply:
yeah, i remember the unused warn was there when we had a debug flag but the debug_assert_eq macro seems working fine.